src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs

```diff
@@ -187,6 +206,151 @@ protected override IEnumerable<Model.AzureSqlManagedInstanceModel> GetEntity()
             return new List<AzureSqlManagedInstanceModel>() { ModelAdapter.GetManagedInstance(this.ResourceGroupName, this.Name) };
         }
 
+        /// <summary>
+        /// Validate requested edition.
+        /// </summary>
+        protected void ValidateRequestedEdition()
+        {
+            ModelAdapter = InitModelAdapter();
+            AzureSqlManagedInstanceModel existingInstance = ModelAdapter.GetManagedInstance(this.ResourceGroupName, this.Name);
+            Management.Internal.Resources.Models.Sku Sku = new Management.Internal.Resources.Models.Sku();
+
+            // Get current edition
+            string currentEdition = existingInstance.Sku.Tier;
+
+            // Check if both SkuName and Edition are set, but editions in them are different
+            if (!string.IsNullOrWhiteSpace(this.SkuName) && !string.IsNullOrWhiteSpace(this.Edition))
+            {
+                if (!this.GetEditionFromSkuName(this.SkuName).Equals(this.Edition, StringComparison.InvariantCultureIgnoreCase))
+                {
+                    throw new PSArgumentException(
+                        string.Format(Microsoft.Azure.Commands.Sql.Properties.Resources.EditionDifferentInSkuAndEdition, this.Edition, this.SkuName),
+                        "Edition");
+                }
+            }
+
+            // Get requested edition
+            string requestedEdition = !string.IsNullOrWhiteSpace(this.SkuName) ? this.GetEditionFromSkuName(this.SkuName) : this.Edition;
+
+            bool isEditionChanged = !string.IsNullOrWhiteSpace(requestedEdition) && !currentEdition.Equals(requestedEdition, StringComparison.InvariantCultureIgnoreCase);
+
+            if (isEditionChanged)
+            {
+                // Check whether managed instance is in instance pool because it is not possible to change hardware family in that case.
+                if (!string.IsNullOrWhiteSpace(existingInstance.InstancePoolName))
+                {
+                    throw new PSArgumentException(
+                        string.Format(Microsoft.Azure.Commands.Sql.Properties.Resources.InstancePoolInstanceCannotChangeHardwareFamily, this.Name, existingInstance.InstancePoolName),
+                        "Edition");
```

Comment:
Why all these validations on client side? Isn't server side doing the same validation? Having business logic on the client side is generally a bad idea because PowerShell/CLI is installed on user's machine, so they need to download updates when our policies change - this is really bad experience.

Reply:
Removed all validation except prompting from confirmation that the changing hardware family is irreversible.